internal/cluster/cluster_test.go

```diff
@@ -3007,8 +3050,13 @@ var _ = Describe("Transform day1 cluster to a day2 cluster", func() {
 				Expect(db.Take(&c, "id = ?", cluster.ID.String()).Error).ToNot(HaveOccurred())
 				Expect(c.Kind).To(Equal(swag.String(models.ClusterKindAddHostsCluster)))
 				Expect(c.Status).To(Equal(swag.String(models.ClusterStatusAddingHosts)))
-				apiVipDnsname := fmt.Sprintf("api.%s.%s", c.Name, c.BaseDNSDomain)
-				Expect(c.APIVipDNSName).To(Equal(swag.String(apiVipDnsname)))
+				if t.apiVip == "" || t.userManagedNetwork {
+					apiVipDnsname := fmt.Sprintf("api.%s.%s", c.Name, c.BaseDNSDomain)
+					Expect(c.APIVipDNSName).To(Equal(swag.String(apiVipDnsname)))
+				} else {
+					Expect(c.APIVipDNSName).To(Equal(swag.String(t.apiVip)))
```

Comment:
I think this demonstrates the intention better:

```suggestion
					Expect(c.APIVipDNSName).To(Equal(swag.String(cluster.APIVip)))
```

Reply:
sounds good